### PR TITLE
Update Istio readme and app-readme

### PR DESCRIPTION
**Pull request by brendarearden**
**Issues**
https://github.com/rancher/rancher/issues/34699
https://github.com/rancher/rancher/issues/33843

**Problem**
When a resource gets deprecated, istio does not always automatically cleanup those resources. So when they deprecated the v1alpha1 security policy and the merging to the istio-policy pod into istiod, upgrades started to fail because manual steps were needed to cleanup to old resources.

**Solution**
Add steps to the app-readme and the readme on how to complete this manual intervention to correct the deprecation of these resources.

### CODE REVIEW

File: index.yaml [hunk 1]
```diff
@@ -1856,6 +1856,39 @@ entries:
     - assets/rancher-grafana/rancher-grafana-6.6.401.tgz
     version: 6.6.401
   rancher-istio:
+  - annotations:
+      catalog.cattle.io/auto-install: rancher-kiali-server-crd=100.0.0+up1.35.0
```
Comment (jiaqiluo) [inline index.yaml:1860]: @brendarearden  Just need a confirmation: the crd chart version can be different from the main chart version, so the value here is valid, right?

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

**Comment by jiaqiluo** [comment]
Is there a workaround if I have already run into this error?

File: packages/rancher-istio/charts/app-readme.md [hunk 2]
```diff
@@ -4,20 +4,65 @@ Our [Istio](https://istio.io/) installer wraps the istioctl binary commands in a
 * **[Kiali](https://kiali.io/)**: Used for graphing traffic flow throughout the mesh
 * **[Jaeger](https://www.jaegertracing.io/)**: A quick start, all-in-one installation used for tracing distributed systemm. This is not production qualified, please refer to jaeger documentation to determine which installation you may need instead.
 
-### Known Issues
+## Known Issues
 
-**Airgapped Environments**
+#### Airgapped Environments
 If you are using this chart in an airgapped environment, you will not be able to upgrade. This is because the `istioctl` upgrade command reaches out to an external repo and it is not configurable. We are tracking the fix for this issue [here](https://github.com/rancher/rancher/issues/33402)
```
Comment (cbron) [inline packages/rancher-istio/charts/app-readme.md:10]: We'll have to remove this when other PR goes in, or adjust it to say the final fix is not in, that issue actually links to long term fix. I'd say we just remove and tell them to use workaround in values.yaml.

File: packages/rancher-istio/charts/README.md [hunk 3]
```diff
@@ -57,13 +69,58 @@ http://tracing.{{ .Values.namespaceOverride }}.svc:{{ .Values.service.externalPo
...
 
-# Installation
+## Known Issues
+
+#### Airgapped Environments
+If you are using this chart in an airgapped environment, you will not be able to upgrade. This is because the `istioctl` upgrade command reaches out to an external repo and it is not configurable. We are tracking the fix for this issue [here](https://github.com/rancher/rancher/issues/33402)
+
+## Chart Deprecations
+
+#### v1alpha1 security policies
```
Comment (cbron) [inline packages/rancher-istio/charts/README.md:85]: This is a ton of info, let's either post this in the rancher docs or the issue, and just link out to them. We want to keep the app-readme clean and small. If istio has these docs available we could also use those.

File: packages/rancher-istio/charts/README.md [hunk 4]
```diff
@@ -1,26 +1,37 @@
-# Rancher Istio Installers
```
Comment (cbron) [inline packages/rancher-istio/charts/README.md:1]: Asking again to see if we can remove this. If not its ok.

**Review by cbron — COMMENTED** [review]
Small nits, then good.

File: index.yaml (hunk 1 above)
Comment (brendarearden) [reply index.yaml:1860]: correct!

File: packages/rancher-istio/charts/README.md (hunk 4 above)
Comment (cbron) [reply packages/rancher-istio/charts/README.md:1]: We should split them up so its not duplicated.

Comment (brendarearden) [reply packages/rancher-istio/charts/README.md:1]: My most recent commit removes duplicate information - I put the info about installing, uninstall and configuration in the readme and deprecations and know issues in the app-readme

File: packages/rancher-istio/charts/README.md (hunk 3 above)
Comment (brendarearden) [reply packages/rancher-istio/charts/README.md]: this has been cleaned up in the most recent commits.

**Comment by brendarearden** [comment]
@jiaqiluo You can use the same steps for migration if you have already attempted to upgrade. I added the error messages you might see if you attempted to upgrade and directions to attempt the migration process to cover this situation.

File: charts/rancher-istio/rancher-istio/100.0.1+up1.10.4/app-readme.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
+# Rancher Istio
+
+Our [Istio](https://istio.io/) installer wraps the istioctl binary commands in a handy helm chart, including an overlay file option to allow complex customization. It also includes:
+* **[Kiali](https://kiali.io/)**: Used for graphing traffic flow throughout the mesh
+* **[Jaeger](https://www.jaegertracing.io/)**: A quick start, all-in-one installation used for tracing distributed systemm. This is not production qualified, please refer to jaeger documentation to determine which installation you may need instead.
```
Comment (cbron) [inline charts/rancher-istio/rancher-istio/100.0.1+up1.10.4/app-readme.md:5]:
```suggestion
* **[Jaeger](https://www.jaegertracing.io/)**: A quick start, all-in-one installation used for tracing distributed system. This is not production qualified, please refer to jaeger documentation to determine which installation you may need instead.
```

File: packages/rancher-istio/charts/README.md [hunk 6]
```diff
@@ -1,40 +1,54 @@
...
+# Rancher-Istio Chart
 
 A Rancher created chart that packages the istioctl binary to install via a helm chart.
```
Comment (cbron) [inline packages/rancher-istio/charts/README.md:3]: lets copy the other one here
```suggestion
Rancher's Istio installer wraps the istioctl binary commands in a handy helm chart, including an overlay file option to allow complex customization.
```

File: packages/rancher-istio/charts/README.md [hunk 7]
```diff
@@ -1,40 +1,54 @@
...
 
 To limit scraping to specific namespaces, set `prometheus.prometheusSpec.ignoreNamespaceSelectors=true` and add one of the following configurations to ensure you can continue to view traffic, metrics and graphs for your deployed resources. 
 
 1. Add a Service Monitor or Pod Monitor in the namespace with the targets you want to scrape.
 1. Add an additionalScrapeConfig to your rancher-monitoring instance to scrape all targets in all namespaces.
 
-####  External Services
+####  Kiali External Services
+
+See the app-readme.md for configuration options.
```
Comment (cbron) [inline packages/rancher-istio/charts/README.md:51]: Is this true ? I don't see them in there, if its the docs link we should just link that here.

File: packages/rancher-istio/charts/app-readme.md [hunk 8]
```diff
@@ -4,47 +4,32 @@ Our [Istio](https://istio.io/) installer wraps the istioctl binary commands in a
...
-
-The Rancher Monitoring app sets `prometheus.prometheusSpec.ignoreNamespaceSelectors=false` which means all namespaces will be scraped by Prometheus by default. This ensures you can view traffic, metrics and graphs for resources deployed in other namespaces.
-
-To limit scraping to specific namespaces, set `prometheus.prometheusSpec.ignoreNamespaceSelectors=true` and add one of the following configurations to ensure you can continue to view traffic, metrics and graphs for your deployed resources.
+## Deprecations
 
-1. Add a Service Monitor or Pod Monitor in the namespace with the targets you want to scrape.
-1. Add an additionalScrapeConfig to your rancher-monitoring instance to scrape all targets in all namespaces.
+#### v1alpha1 security policies
+As of 1.6, Istio deprecated support for `v1alpha1` security policies resource and replaced the API with `v1beta1` authorization policies. https://istio.io/latest/docs/reference/config/security/authorization-policy/
```
Comment (cbron) [inline packages/rancher-istio/charts/app-readme.md:16]: Isn't it not deprecated but actually removed ? I thought deprecation happened earlier.

File: packages/rancher-istio/charts/app-readme.md [hunk 9]
```diff
@@ -4,47 +4,32 @@ Our [Istio](https://istio.io/) installer wraps the istioctl binary commands in a
...
 
-This url depends on the values for `namespaceOverride`, and `.Values.service.externalPort` in your rancher-tracing or other tracing instance.:
-```
-http://tracing.{{ .Values.namespaceOverride }}.svc:{{ .Values.service.externalPort }}/{{ .Values.tracing.contextPath }}
-```
+##### Migrating Resources:
+Migration steps can be found in this [istio blog post](https://istio.io/latest/blog/2021/migrate-alpha-policy/ "istio blog post"). 
 
-For more information on how to use the feature, refer to our [docs](https://rancher.com/docs/rancher/v2.x/en/istio/v2.5/).
+You can also use these [quick steps](http://https://github.com/rancher/rancher/issues/33843#issuecomment-921868909 "quick steps") to determine if you need to follow the more extensive migration steps.
```
Comment (cbron) [inline packages/rancher-istio/charts/app-readme.md:35]: bad link

**Review by cbron — CHANGES_REQUESTED** [review]

File: charts/rancher-istio/rancher-istio/100.0.1+up1.10.4/app-readme.md (hunk 5 above)
Comment (cbron) [reply charts/rancher-istio/rancher-istio/100.0.1+up1.10.4/app-readme.md:5]: Bah, I commented on the charts on some of these instead of the package.

**Review by jiaqiluo — DISMISSED** [review]
I just wonder if it is worth releasing a new (patch) version if there are no code changes/ bug fixes. 

However, fixing the documentation is also a kind of bug fixing, so I am totally fine to release a new version.

File: packages/rancher-istio/charts/app-readme.md (hunk 9 above)
Comment (brendarearden) [reply packages/rancher-istio/charts/app-readme.md]: link has been updated in latest commits

File: packages/rancher-istio/charts/app-readme.md (hunk 8 above)
Comment (brendarearden) [reply packages/rancher-istio/charts/app-readme.md]: you are correct - removed not deprecated - I have updated the wording in the latest commits

File: packages/rancher-istio/charts/README.md (hunk 7 above)
Comment (brendarearden) [reply packages/rancher-istio/charts/README.md]: this was left behind when I was moving stuff around - good catch - it has been removed in the latest commits.

File: packages/rancher-istio/charts/README.md (hunk 6 above)
Comment (brendarearden) [reply packages/rancher-istio/charts/README.md]: I have copied this over in the latest commits.

File: charts/rancher-istio/rancher-istio/100.0.1+up1.10.4/app-readme.md (hunk 5 above)
Comment (brendarearden) [reply charts/rancher-istio/rancher-istio/100.0.1+up1.10.4/app-readme.md]: fixed spelling in latest commits.

**Comment by brendarearden** [comment]
@cbron how do you feel about @jiaqiluo comment on the versioning - should we just replaced the existing 100.0.0 version since we are only making readme updates?

**Comment by cbron** [comment]
It doesn't matter that much, but technically we aren't supposed to update existing charts outside of the rancher min/max, and updating released charts is painful/error-prone, so new version it is. 
Just make sure it can upgrade seamlessly (because istio is same version). 
also needs to get added here: https://github.com/rancher/rancher/issues/34589

**Review by cbron — APPROVED** [review]

**Review by jiaqiluo — APPROVED** [review]

**Pull request merged**